agent-tooling/src/main/java/io/opentelemetry/auto/typed/span/HttpClientTypedSpan.java

```diff
@@ -0,0 +1,20 @@
+package io.opentelemetry.auto.typed.span;
+
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+
+public abstract class HttpClientTypedSpan<T extends HttpClientTypedSpan, REQUEST, RESPONSE>
+    extends ClientTypedSpan<T, REQUEST, RESPONSE> {
+
+  public HttpClientTypedSpan(Tracer tracer, Span delegate) {
+    super(tracer, delegate);
+  }
+
+  public T onRequest(REQUEST request) {
+    tracer.getHttpTextFormat().inject(getContext(), request, getSetter());
```

Comment:
Can this be moved to the tracer? (similar to how extraction is done in the `HttpServerTypedTracer` below)

Added benefit is then I don't think the `tracer` field in `BaseTypedSpan` would be needed.

Reply:
done.